lib/dialects/postgres/query/compiler.js

```diff
@@ -26,8 +26,11 @@ class QueryCompiler_PG extends QueryCompiler {
 
     const { returning, onConflict, ignore, merge, insert } = this.single;
     if (onConflict && ignore) sql += this._ignore(onConflict);
-    if (onConflict && merge)
+    if (onConflict && merge) {
       sql += this._merge(merge.updates, onConflict, insert);
+      const wheres = this.where();
+      sql += wheres ? ` ${wheres}` : '';
```

Comment:
I would simlify this to a `if (wheres)` check instead of empty concatenation, same as here:
`if (returning) sql += this._returning(returning);`

Reply:
Will do

Reply:
706ce6e42513c52547c045827a504a32991e941b